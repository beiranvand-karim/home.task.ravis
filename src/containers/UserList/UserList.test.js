import React from 'react'
import {shallow} from "enzyme"
import renderer from 'react-test-renderer'
import {MemoryRouter} from 'react-router-dom'
import {UserList} from "./"

describe('UserList Component', () => {
   const mockDeleteUser = jest.fn();
   const mockHistory = { push: jest.fn() };
   const users = [
      {
         name: 'somevalue',
         family: 'somevalue',
         nickName: 'somevalue',
         phoneNumber: 'somevalue',
         email: 'somevalue',
         community: 'somevalue',
         birthDay: 'somevalue',
         gender: 'somevalue'
      }
   ];

   it('should call delete user', () => {
      const wrapper = shallow(<UserList users={users} deleteUser={mockDeleteUser}/>);
      wrapper.find('.deleteUser').simulate('click');
      expect(mockDeleteUser.mock.calls.length).toEqual(1)
   });

   it('should push to history', () => {
      const wrapper = shallow(<UserList users={users} history={mockHistory}/>);
      wrapper.find('.pushHistory').simulate('click');
      expect(mockHistory.push.mock.calls[0]).toEqual([`/updateUser/${0}`])
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><UserList users={users}/></MemoryRouter>);
      expect(tree).toMatchSnapshot()
   })
});
