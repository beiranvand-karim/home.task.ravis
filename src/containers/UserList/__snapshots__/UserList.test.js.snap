// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList Component should matches the snapshot 1`] = `
<div
  className="userListContainer"
>
  <div
    className="userListHeader"
  >
    <div
      className="userListHeader userListHeader__row"
    >
      name
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      family
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      nickName
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      p. Number
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      email
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      community
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      birthDay
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      gender
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      edit
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      delete
    </div>
  </div>
  <div
    className="userListHeader"
  >
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      somevalue
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      <button
        className="pushHistory"
        onClick={[Function]}
      >
        edit
      </button>
    </div>
    <div
      className="userListHeader userListHeader__row"
    >
      <button
        className="deleteUser"
        onClick={[Function]}
      >
        delete
      </button>
    </div>
  </div>
</div>
`;
